algebird-core/src/main/scala/com/twitter/algebird/QTree.scala

```diff
@@ -63,8 +61,26 @@ object QTree {
   def apply(k: Double): QTree[Double] = apply(k -> k)
 }
 
-class QTreeSemigroup[A: Monoid](k: Int) extends Semigroup[QTree[A]] {
+class QTreeSemigroup[A](k: Int)(implicit val underlyingMonoid: Monoid[A]) extends Semigroup[QTree[A]] {
```

Comment:
That's what implicitly is for, right?  The main benefit of having is as a parameter is that it doesn't have to be implicitly specified.